lua/triptych/types.lua

```diff
@@ -73,10 +73,18 @@
 ---@field signs TriptychConfigGitSignsSigns
 
 ---@class TriptychConfigGitSignsSigns
----@field add string
----@field modify string
----@field rename string
----@field untracked string
+---@field add string | TriptychConfigGitSignDefineOptions
```

Comment:
Given that we're not referencing any of the properties in `TriptychConfigGitSignDefineOptions` I wonder if there's anything gained from typing it, rather than just doing `string | table`? 

When you're writing your Triptych config to use to new table options, does the language server give you autocomplete hints about the properties?

Reply:
Actually, the types can be visible! This part of my config allows me to get lsp support for the neovim API (sans the parts written in C, but they said that it would be fixed in 10.0).

```lua
require("lspconfig").lua_ls.setup({
    settings = {
        Lua = {
            runtime = {
                -- Tell the language server which version of Lua you're using (most likely LuaJIT in the case of Neovim)
                version = "LuaJIT",
            },
            diagnostics = {
                -- Get the language server to recognize the `vim` global
                globals = { "vim" },
            },
            workspace = {
                -- Make the server aware of Neovim runtime files
                library = vim.api.nvim_get_runtime_file("", true),
            },
            -- Do not send telemetry data containing a randomized but unique identifier
            telemetry = {
                enable = false,
            },
        },
    },
})
```